networkit/cpp/community/test/CommunityGTest.cpp

```diff
@@ -174,6 +175,28 @@ TEST_F(CommunityGTest, testPLM) {
 
 }
 
+TEST_F(CommunityGTest, testParallelLeiden) {
+    METISGraphReader reader;
+    Modularity modularity;
+    Graph G = reader.read("input/PGPgiantcompo.graph");
+
+    ParallelLeiden pl(G);
+    pl.run();
+    Partition zeta = pl.getPartition();
+
+    DEBUG("number of clusters: " , zeta.numberOfSubsets());
+    DEBUG("modularity: " , modularity.getQuality(zeta, G));
+    EXPECT_TRUE(GraphClusteringTools::isProperClustering(G, zeta));
+
+    ParallelLeiden plnr(G,3,false);
+    plnr.run();
+    Partition zeta2 = plnr.getPartition();
+
+    DEBUG("number of clusters: " , zeta2.numberOfSubsets());
+    DEBUG("modularity: " , modularity.getQuality(zeta2, G));
+    EXPECT_TRUE(GraphClusteringTools::isProperClustering(G, zeta2));
+}
```

Comment:
The algorithm seems quite complex but you have just one test. Does this cover all the relevant parts of the algorithm?

Reply:
Not at all. I used to "test" my algorithm simply by comparing the results and whether or not the algorithm crashed. 
I didn't really come up with any good ideas to test individual parts of it due to its non-deterministic behavior. It's also unclear when a result is found to be good or not, since higher modularity doesn't necessarily mean a better result. I also had no necessity for them, as my code isn't even evaluated as part of my Bachelor thesis.

The only other simple tests I can come up with on the spot is checking that every node is assigned to a community. One could also check that every node in the refined partition is still within its host community, which would be a little more effort though.

Reply:
For non-deterministic algos, it's OK to simply test that they don't crash.

If possible (in reasonable time), it would be nice if the "algo does not crash" tests are exhaustive though (= all cases of the algorithm are covered). Does the current test trigger all code paths (more-or-less)?

Reply:
No, the resize logic is only triggered if the shared communityVolumes vector becomes too small. This doesn't even happen on the largest graph I  found (3b edges, 100m nodes) so it'd require making the default oversizing of the vectors smaller for the test specifically.

Since im not aware of any methods to change private variables (like reflection in Java) this would mean making the oversize variable public / adding a setter.

Apart from that every line of code should be triggered, unless the graph is small enough to make the algorithm stop after a single local moving.

Reply:
Making the variables public and writable sounds fine to me.

Reply:
Do you want me to make the other constexpr variable public then as well? (The one WORKING_SIZE one. Specifies how many nodes are collected in a thread until the vector is pushed to a shared queue)

Reply:
You can probably just expose whatever is necessary for the test (unless that leads to a very ugly interface or so). If there are no invariants (or only straightforward ones that can be checked at the top of `run()`), I'd prefer public members over setters + getters.